### ISSUE TITLE
[BUG] Crash after turning shader on

### ISSUE THREAD

**Issue by DanMeadWasTaken** [gri573/rethinking-voxels#46]
I wasn't sure where else to go about this? So, after I turned on the shaderspack, my entire prism client and Minecraft client just instantly crashed, no logs. then after I tried joining a server with it technically still on, client crashed and got this log. any suggestions?

https://mclo.gs/JdACDQt

**Comment by gri573** [comment]
Thanks, that's a bug and will hopefully be fixed in the next version even if I can't reproduce it

**Comment by DanMeadWasTaken** [comment]
Well, i think it might have something to do with indium? since the logs said it had to do with sodium and indium helps modify sodium inorder to make it compatible with mods? as i assume that most people use sodium, and don't have the same issue. let me try again with sodium disabled- although either way, it'd be nice to get it fixed

**Comment by gri573** [comment]
I have used this shaderpack without problems with indium, the error says that extension statements are not possible in the middle of shaders,and i did in fact put an extension statement in a place that isn't the very start of the relevant shader program. This is most likely caused by your graphics driver enforcing the GLSL spec more strictly than most others in this case. If you want to test that, you can move the line `#extension [some name I don't fully remember] : enable` from `shaders/program/shadow_fsh.glsl` the top of `shaders/world*/shadow.fsh` (* being wildcard, but i think you know that given your ability to provide logs/be generally helpful while reporting this)

**Comment by gri573** [comment]
File paths are relative to shaderpack root in my last comment

**Comment by DanMeadWasTaken** [comment]
so, just to make sure, you want me to move this line at the top to shaders/world*/shadow.fsh right?

![image](https://user-images.githubusercontent.com/75497984/216297421-f5dfb5aa-d355-4477-8627-68761c6754bf.png)

**Comment by gri573** [comment]
Yes

**Comment by gri573** [comment]
I meant adding it to all of those files, each of those folders contains the shader files for the respective versions (it should be world-1, world0 and world1), so only one of them is loaded at a time

**Comment by DanMeadWasTaken** [comment]
Well- it's a good thing i tested before seeing this anyway lol

world0 - doesn't work
world1 - does work but everything is absurdly white and drop from 60fps to 2-3fps
world-1 works as intended but again massive frame drop

when in all 3, doesn't work

**Comment by gri573** [comment]
That's a pity, maybe iris injects things at the start of the shader, i'll ask their dev team about it. A thing that should work is deleting the line completely and not enabling instant light propagation. Performance shouldn't be that bad, on low or medium profile it should be about 15-25 if you are playing with complementary at that framerate in 1080p

**Comment by gri573** [comment]
the mod called chunksfadein is apparently mixing into iris and adds code before the start of shader programs, removing that should fix the crash.

**Comment by zndrmn** [comment]
hmm I'm using chunksfadein and I haven't had any problems yet

**Comment by gri573** [comment]
yes, because not every driver needs to have extension declarations at the start of the shader
